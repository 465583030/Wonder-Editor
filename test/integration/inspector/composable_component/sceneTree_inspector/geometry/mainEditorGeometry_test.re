open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

open MainEditorTransform;

let _ =
  describe("MainEditorGeometry component", () => {
    let sandbox = getSandboxDefaultVal();
    beforeEach(() => {
      sandbox := createSandbox();

      MainEditorSceneTool.initState(~sandbox, ());
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    describe("test set currentSceneTreeNode", () => {
      beforeEach(() => {
        MainEditorSceneTool.createDefaultScene(
          sandbox,
          MainEditorSceneTool.setFirstBoxTobeCurrentSceneTreeNode,
        );

        CurrentSelectSourceEditorService.setCurrentSelectSource(
          EditorType.SceneTree,
        )
        |> StateLogicService.getAndSetEditorState;
      });

      describe("test change geometry", () => {
        describe("test snapshot", () => {
          test("test show select geometry group widget", () => {
            let currentGameObjectGeometry =
              GameObjectTool.getCurrentGameObjectGeometry();

            let component =
              BuildComponentTool.buildGeometry(
                TestTool.buildEmptyAppState(),
                currentGameObjectGeometry,
              );

            BaseEventTool.triggerComponentEvent(
              component,
              MainEditorGeometryTool.triggerClickShowGeometryGroup,
            );

            component |> ReactTestTool.createSnapshotAndMatch;
          });
          test("test hide select geometry group widget", () => {
            let currentGameObjectGeometry =
              GameObjectTool.getCurrentGameObjectGeometry();

            let component =
              BuildComponentTool.buildGeometry(
                TestTool.buildEmptyAppState(),
                currentGameObjectGeometry,
              );

            BaseEventTool.triggerComponentEvent(
              component,
              MainEditorGeometryTool.triggerClickShowGeometryGroup,
            );
            BaseEventTool.triggerComponentEvent(
              component,
              MainEditorGeometryTool.triggerClickHideGeometryGroup,
            );

            component |> ReactTestTool.createSnapshotAndMatch;
          });
        });

        describe("test logic", () => {
          test("test the current gameObject geometry should is Cube", () => {
            let currentGameObjectGeometry =
              GameObjectTool.getCurrentGameObjectGeometry();

            GeometryEngineService.getGeometryName(currentGameObjectGeometry)
            |> StateLogicService.getEngineStateToGetData
            |> expect == "Cube";
          });
          test(
            "test change geometry to be Sphere, the current gameObject geometry should is Sphere",
            () => {
              let component =
                BuildComponentTool.buildGeometry(
                  TestTool.buildEmptyAppState(),
                  GameObjectTool.getCurrentGameObjectGeometry(),
                );

              BaseEventTool.triggerComponentEvent(
                component,
                MainEditorGeometryTool.triggerClickShowGeometryGroup,
              );

              BaseEventTool.triggerComponentEvent(
                component,
                MainEditorGeometryTool.getSphereDomIndex()
                |> MainEditorGeometryTool.triggerClickSpecificGeometry,
              );

              let newGameObjectGeometry =
                GameObjectTool.getCurrentGameObjectGeometry();

              GeometryEngineService.getGeometryName(newGameObjectGeometry)
              |> StateLogicService.getEngineStateToGetData
              |> expect == "Sphere";
            },
          );
        });
      });
    });
  });