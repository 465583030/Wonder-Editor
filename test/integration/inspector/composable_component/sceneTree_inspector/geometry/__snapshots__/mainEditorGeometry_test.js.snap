// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainEditorGeometry component test set currentSceneTreeNode test change geometry test snapshot test hide select geometry group widget 1`] = `
<article
  className="wonder-inspector-geometry"
>
  <div
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Geometry
    </div>
    <div
      className="item-content"
    >
      <div
        className="inspector-select"
      >
        <div
          className="select-name"
        >
          Cube
        </div>
        <div
          className="select-img"
          onClick={[Function]}
        >
          <img
            src="./public/img/select.png"
          />
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`MainEditorGeometry component test set currentSceneTreeNode test change geometry test snapshot test show select geometry group widget else, select geometry group should only contain geometry which has texCoord 1`] = `
<article
  className="wonder-inspector-geometry"
>
  <div
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Geometry
    </div>
    <div
      className="item-content"
    >
      <div
        className="inspector-select"
      >
        <div
          className="select-name"
        >
          Cube
        </div>
        <div
          className="select-img"
          onClick={[Function]}
        >
          <img
            src="./public/img/select.png"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="select-component-content"
  >
    <div
      className="select-component-item"
    >
      <div
        className="item-header"
      >
        Geometry
      </div>
      <div
        className="item-content item-active"
        onClick={[Function]}
      >
        Cube
      </div>
      <div
        className="item-content"
        onClick={[Function]}
      >
        Sphere
      </div>
    </div>
    <div
      className="select-component-bg"
      onClick={[Function]}
    />
  </div>
</article>
`;

exports[`MainEditorGeometry component test set currentSceneTreeNode test change geometry test snapshot test show select geometry group widget if current geometry has no texCoord, select geometry group should contain geometry which has texCoord or no texCoord 1`] = `
<article
  className="wonder-inspector-geometry"
>
  <div
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Geometry
    </div>
    <div
      className="item-content"
    >
      <div
        className="inspector-select"
      >
        <div
          className="select-name"
        >
          Cube
        </div>
        <div
          className="select-img"
          onClick={[Function]}
        >
          <img
            src="./public/img/select.png"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="select-component-content"
  >
    <div
      className="select-component-item"
    >
      <div
        className="item-header"
      >
        Geometry
      </div>
      <div
        className="item-content item-active"
        onClick={[Function]}
      >
        Cube
      </div>
      <div
        className="item-content"
        onClick={[Function]}
      >
        Sphere
      </div>
      <div
        className="item-content"
        onClick={[Function]}
      >
        geometryNoTexCoord
      </div>
    </div>
    <div
      className="select-component-bg"
      onClick={[Function]}
    />
  </div>
</article>
`;

exports[`MainEditorGeometry component test set currentSceneTreeNode test load asset wdb test select geometry group widget should show all geometry 1`] = `
<article
  className="wonder-inspector-geometry"
>
  <div
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Geometry
    </div>
    <div
      className="item-content"
    >
      <div
        className="inspector-select"
      >
        <div
          className="select-name"
        >
          Cube
        </div>
        <div
          className="select-img"
          onClick={[Function]}
        >
          <img
            src="./public/img/select.png"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="select-component-content"
  >
    <div
      className="select-component-item"
    >
      <div
        className="item-header"
      >
        Geometry
      </div>
      <div
        className="item-content item-active"
        onClick={[Function]}
      >
        Cube
      </div>
      <div
        className="item-content"
        onClick={[Function]}
      >
        Sphere
      </div>
      <div
        className="item-content"
        onClick={[Function]}
      >
        Mesh
      </div>
    </div>
    <div
      className="select-component-bg"
      onClick={[Function]}
    />
  </div>
</article>
`;
